tools/ping_enumerator.py

```diff
@@ -3,76 +3,145 @@
-        Enumerate new ping devices
-    """
-    output = ""
-    # Remove any previously created links
-    try:
-        output = subprocess.check_output(["rm", "-rf", "/dev/serial/ping"])
-    except subprocess.CalledProcessError as exception:
-        print(exception)
+class PingEnumerator:
 
-    # Look for connected serial devices
-    try:
-        output = subprocess.check_output("ls /dev/serial/by-id", shell=True)
-    except subprocess.CalledProcessError as exception:
-        print(exception)
-        exit(1)
+    def legacy_detect_ping1d(self, ping):
+        """
+        Detects Ping1D devices without DEVICE_INFORMATION implemented
+        """
+        firmware_version = ping.request(PING1D_FIRMWARE_VERSION)
+        if firmware_version is None  or not hasattr(firmware_version, "device_type"):
+            return None
+        description = "/dev/serial/ping/Ping1D-id-%s-t-%s-m-%s-v-%s.%s" % (
+            firmware_version.src_device_id,
+            firmware_version.device_type,
+            firmware_version.device_model,
+            firmware_version.firmware_version_major,
+            firmware_version.firmware_version_minor
+        )
+        return description
 
-    pings_found = 0
+    def detect_device(self, dev):
+        """
+        Attempts to detect the Ping device attached to serial port 'dev'
+        Returns the new path with encoded name if detected, or None if the
+        device was not detected
+        """
 
-    # Look at each serial device, probe for ping
-    for dev in output.split('\n'):
-        if not dev:
-            continue
+        ping = PingDevice("/dev/serial/by-id/" + dev, 115200)
+        if not ping.initialize():
+            return None
 
-        byiddev = "/dev/serial/by-id/" + dev
+        device_info = ping.request(COMMON_DEVICE_INFORMATION)
+        if not device_info or not hasattr(device_info, "device_type"):
+            return self.legacy_detect_ping1d(ping)
 
-        print("Looking for Ping at %s" % byiddev)
-        new_ping = Ping1D(byiddev)
+        if device_info.device_type == 1:
+            description = "/dev/serial/ping/Ping1D-id-%s-r-%s-v-%s.%s.%s"
+        elif device_info.device_type == 2:
+            description = "/dev/serial/ping/Ping360-id-%s-r-%s-v-%s.%s.%s"
+            # Open device with 2M baud to setup Ping360
+            print("Setting baud to 2M...")
+            ser = serial.Serial("/dev/serial/by-id/" + dev, 2000000)
+            ser.send_break()
+            ser.write("UUUUUUU")
+            ser.close()
+            self.set_low_latency(dev)
 
-        if not new_ping.initialize():
-            continue
+        else:
+            return None
 
+        return description % (
+            device_info.src_device_id,
+            device_info.device_revision,
+            device_info.firmware_version_major,
+            device_info.firmware_version_minor,
+            device_info.firmware_version_patch
+        )
+
+    def set_low_latency(self, dev):
```

Comment:
should this make it's way into ping-python device class?

Reply:
does not block pr

Reply:
> @Williangalvani I am working on testing this on pi now. Can you work on adding the install step?

Yep